src/s390x/zipl.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::dasd::check;
+use crate::errors::*;
+use regex::Regex;
+use std::fs::read_to_string;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tempfile::NamedTempFile;
+use walkdir::WalkDir;
+
+/////////////////////////////////////////////////////////////////////////////
+// IBM Z Bootloader Support
+/////////////////////////////////////////////////////////////////////////////
+
+/// Generate zipl config and set dasd as boot device
```

Comment:
Hmm interesting.  We have code for this in coreos-assembler too.  One thing perhaps we could do is factor out a hidden `coreos-installer internals s390x-update-zipl` command to share it?

Reply:
Hmm. Maybe in the future? I don't have a strong opinion about that